 ## DATASET DE VACUNACION 
 This dataset has 264 patients and 8 different features or columns and from a visual inspection we can find that there are two long description columns, one refering to the health service and the second to the diagnostic. 

In [1]:
import pandas as pd
import numpy as np
import missingno as msno
import matplotlib.pyplot as plt
import seaborn as sns
import folium
import json
import requests
import geopandas as gpd
%matplotlib inline

In [2]:
vaccination = pd.read_excel(f'./Data/Vacunacion.xlsx')
#dyspnea = pd.read_excel(f'./Data/Disnea.xlsx')
vaccination

Numero_de_documento Fecha_Emision Regional_EPS_Desc Codigo_Prestacion_Op  \
0                 845676    2018-01-29            Bogota             S1120191   
1                 686768    2018-05-20          Medellin             S1120191   
2                1790062    2018-06-05          Medellin             S1120191   
3                1142931    2018-01-10          Medellin             S1120191   
4                1734917    2018-06-06          Medellin             S1120191   
..                   ...           ...               ...                  ...   
259               695889    2020-05-06          Medellin             S1120192   
260               682268    2020-05-13          Medellin             S1120191   
261              1932263    2020-05-11            Bogota             S1120191   
262              1233344    2020-05-14          Medellin             S1120195   
263              1035965    2020-05-13          Medellin             S1120192   

                              Descripcion_Prestacion  \
0    VACUNA PAI INFLUENZA NIÑOS/AS MAYORES DE 9 AÑOS   
1    VACUNA PAI INFLUENZA NIÑOS/AS MAYORES DE 9 AÑOS   
2    VACUNA PAI INFLUENZA NIÑOS/AS MAYORES DE 9 AÑOS   
3    VACUNA PAI INFLUENZA NIÑOS/AS MAYORES DE 9 AÑOS   
4    VACUNA PAI INFLUENZA NIÑOS/AS MAYORES DE 9 AÑOS   
..                                               ...   
259  VACUNA PAI INFLUENZA NIÑOS/AS MAYORES DE 9 AÑOS   
260  VACUNA PAI INFLUENZA NIÑOS/AS MAYORES DE 9 AÑOS   
261  VACUNA PAI INFLUENZA NIÑOS/AS MAYORES DE 9 AÑOS   
262  VACUNA PAI INFLUENZA NIÑOS/AS MAYORES DE 9 AÑOS   
263  VACUNA PAI INFLUENZA NIÑOS/AS MAYORES DE 9 AÑOS   

    Codigo_Diagnostico_EPS_Op  \
0                        Z251   
1                        Z759   
2                        Z251   
3                        Z759   
4                        Z251   
..                        ...   
259                      Z251   
260                      Z759   
261                      Z251   
262                      Z251   
263                      Z251   

                                  Diagnostico_EPS_Desc  Cantidad_Autorizada  
0    NECESIDAD DE INMUNIZACIÓN CONTRA LA INFLUENZA ...                    1  
1    PROBLEMA NO ESPECIFICADO RELACIONADO CON SERVI...                    1  
2    NECESIDAD DE INMUNIZACIÓN CONTRA LA INFLUENZA ...                    1  
3    PROBLEMA NO ESPECIFICADO RELACIONADO CON SERVI...                    1  
4    NECESIDAD DE INMUNIZACIÓN CONTRA LA INFLUENZA ...                    1  
..                                                 ...                  ...  
259  NECESIDAD DE INMUNIZACIÓN CONTRA LA INFLUENZA ...                    1  
260  PROBLEMA NO ESPECIFICADO RELACIONADO CON SERVI...                    1  
261  NECESIDAD DE INMUNIZACIÓN CONTRA LA INFLUENZA ...                    1  
262  NECESIDAD DE INMUNIZACIÓN CONTRA LA INFLUENZA ...                    1  
263  NECESIDAD DE INMUNIZACIÓN CONTRA LA INFLUENZA ...                    1  

[264 rows x 8 columns]

So we can count the amount of people that is currently under this enterprise by selecting using the function `reset_index()loc` to determine the amount of people thatcan be found by city. Therefore from the table we just generated below we can see that most of the population corresponds to Medellin areea 

In [4]:
vaccination[['Fecha_Emision']].describe()

Fecha_Emision
count                   264
unique                  170
top     2019-06-05 00:00:00
freq                      6
first   2018-01-03 00:00:00
last    2020-05-14 00:00:00

From this data we can see that most of the measurements have been performed between 2018 and 2020

In [5]:
loc = vaccination[['Numero_de_documento','Regional_EPS_Desc']].groupby('Regional_EPS_Desc').count().reset_index()
loc

Regional_EPS_Desc  Numero_de_documento
0      Barranquilla                   11
1            Bogota                   16
2              Cali                   12
3         Manizales                    4
4          Medellin                  219
5           Pereira                    2

In [6]:
vaccination.groupby(['Numero_de_documento'],as_index=False).count()

Numero_de_documento  Fecha_Emision  Regional_EPS_Desc  \
0                 500547              2                  2   
1                 537607              2                  2   
2                 539998              1                  1   
3                 540250              2                  2   
4                 541007              1                  1   
..                   ...            ...                ...   
156              1932263              4                  4   
157              1940079              2                  2   
158              1971217              1                  1   
159              1973217              1                  1   
160              1985014              2                  2   

     Codigo_Prestacion_Op  Descripcion_Prestacion  Codigo_Diagnostico_EPS_Op  \
0                       2                       2                          2   
1                       2                       2                          2   
2                       1                       1                          1   
3                       2                       2                          2   
4                       1                       1                          1   
..                    ...                     ...                        ...   
156                     4                       4                          4   
157                     2                       2                          2   
158                     1                       1                          1   
159                     1                       1                          1   
160                     2                       2                          2   

     Diagnostico_EPS_Desc  Cantidad_Autorizada  
0                       2                    2  
1                       2                    2  
2                       1                    1  
3                       2                    2  
4                       1                    1  
..                    ...                  ...  
156                     4                    4  
157                     2                    2  
158                     1                    1  
159                     1                    1  
160                     2                    2  

[161 rows x 8 columns]

By grouping the dataset by identification numbers, we have identified that there are 161 patients and therefore there are 103 repetitions


In [7]:
msno.matrix(msno.nullity_sort(vaccination, sort='descending'))
# This dataset seems to be complete

NameError: name 'msno' is not defined

In [8]:
fig, ax = plt.subplots(figsize=(14,10))
vaccination['Regional_EPS_Desc'].hist(bins=8)

NameError: name 'plt' is not defined

In [9]:
vaccination.groupby(['Descripcion_Prestacion'],as_index=False).count().head(20)

Descripcion_Prestacion  Numero_de_documento  \
0         HEPATITIS A, ANTICUERPOS IG M [ANTI HVA-M]                    1   
1        HEPATITIS A, ANTICUERPOS TOTALES [ANTI HVA]                    1   
2     HEPATITIS B, ANTICUERPOS SUPERFICIE [ANTI-HBS]                   15   
3  VACUNA HEPATITIS B ADULTOS -INDICACIONES DIFER...                    1   
4  VACUNA INFLUENZA PAI MUJERES GESTANTES A PARTI...                    1   
5  VACUNA NEUMOCOCO 23 -USO EXCLUSIVO BOGOTA Y CA...                    2   
6    VACUNA PAI INFLUENZA NIÑOS/AS MAYORES DE 9 AÑOS                  242   
7    VACUNA PAI INFLUENZA NIÑOS/ASHASTA LOS 23 MESES                    1   

   Fecha_Emision  Regional_EPS_Desc  Codigo_Prestacion_Op  \
0              1                  1                     1   
1              1                  1                     1   
2             15                 15                    15   
3              1                  1                     1   
4              1                  1                     1   
5              2                  2                     2   
6            242                242                   242   
7              1                  1                     1   

   Codigo_Diagnostico_EPS_Op  Diagnostico_EPS_Desc  Cantidad_Autorizada  
0                          1                     1                    1  
1                          1                     1                    1  
2                         15                    15                   15  
3                          1                     1                    1  
4                          1                     1                    1  
5                          2                     2                    2  
6                        242                   242                  242  
7                          1                     1                    1

In [10]:
vaccination['Descripcion_Prestacion'].astype('category')
vaccination['Diagnostico_EPS_Desc'].astype('category')

0      NECESIDAD DE INMUNIZACIÓN CONTRA LA INFLUENZA ...
1      PROBLEMA NO ESPECIFICADO RELACIONADO CON SERVI...
2      NECESIDAD DE INMUNIZACIÓN CONTRA LA INFLUENZA ...
3      PROBLEMA NO ESPECIFICADO RELACIONADO CON SERVI...
4      NECESIDAD DE INMUNIZACIÓN CONTRA LA INFLUENZA ...
                             ...                        
259    NECESIDAD DE INMUNIZACIÓN CONTRA LA INFLUENZA ...
260    PROBLEMA NO ESPECIFICADO RELACIONADO CON SERVI...
261    NECESIDAD DE INMUNIZACIÓN CONTRA LA INFLUENZA ...
262    NECESIDAD DE INMUNIZACIÓN CONTRA LA INFLUENZA ...
263    NECESIDAD DE INMUNIZACIÓN CONTRA LA INFLUENZA ...
Name: Diagnostico_EPS_Desc, Length: 264, dtype: category
Categories (15, object): [ARTRITIS REUMATOIDE, NO ESPECIFICADA, ASMA, NO ESPECIFICADA, DEGENERACIÓN GRASA DEL HÍGADO, NO CLASIFICADA ..., ENTESOPATÍA DEL MIEMBRO INFERIOR, NO ESPECIFICADA, ..., OTRAS CIRROSIS DEL HÍGADO Y LAS NO ESPECIFICADAS, OTRAS URTICARIAS, OTROS EXÁMENES GENERALES, PROBLEMA NO ESPECIFICADO REL

In [11]:
vaccination.isnull().any()

Numero_de_documento          False
Fecha_Emision                False
Regional_EPS_Desc            False
Codigo_Prestacion_Op         False
Descripcion_Prestacion       False
Codigo_Diagnostico_EPS_Op    False
Diagnostico_EPS_Desc         False
Cantidad_Autorizada          False
dtype: bool